src/software/geom/algorithms/rasterize.cpp

```diff
@@ -0,0 +1,293 @@
+#include "software/geom/algorithms/rasterize.h"
+
+#include <algorithm>
+#include <cmath>
+
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/algorithms/distance.h"
+#include "software/geom/algorithms/intersection.h"
+#include "software/geom/algorithms/rasterize.h"
+
+
+bool isInPixel(const Point& a, const Point& b, double resolution_size);
+std::optional<Point> isAVertex(const Point& point, const Polygon& polygon, double resolution_size);
+const Point& find_next_point_in_polygon(const Point& point, const Polygon& polygon);
+const Point& find_previous_point_in_polygon(const Point& point, const Polygon& polygon);
+bool is_special_vertex(const Point& point, const Polygon& polygon, const double resolution_size);
+
+std::vector<Point> rasterize(const Circle& circle, const double resolution_size)
+{
+    std::vector<Point> covered_points;
+
+    // Using an approach to find the points on the edges using y = +-sqrt(r^2 - (x - k)^2)
+    // + h and filling in the rest Downside is using sqrt to calculate Chosen to avoid
+    // points being outside of circle due to floating point errors
```

Comment:
is there a link to a description of this algorithm and similarly for the rest of the rasterize functions?

Reply:
These algorithms don't really have any links to support them as they are all modifications of brute forcing. I have updated the comments to be more descriptive.